applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -634,7 +640,11 @@ static bool key_value_set(struct items *items, u16_t usage_id, s16_t value)
 
 static void send_report_keyboard(u8_t report_id, struct report_data *rd)
 {
-	__ASSERT_NO_MSG(report_id == REPORT_ID_KEYBOARD_KEYS);
+	__ASSERT_NO_MSG((IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_KEYBOARD_SUPPORT) &&
+			 (report_id == REPORT_ID_KEYBOARD_KEYS)) ||
+			(IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_KEYBOARD) &&
+			 (report_id == REPORT_ID_BOOT_KEYBOARD)));
+	/* Both normal and boot protocol reports use the same formatting. */
 
 	if (!IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_KEYBOARD_SUPPORT)) {
```

Comment:
In case we get REPORT_ID_BOOT_KEYBOARD and CONFIG_DESKTOP_HID_BOOT_INTERFACE_KEYBOARD is disabled we will still execute code in release configurations.

Reply:
there is a dependency in kconfig - I would leave it